internal/task/task/common/get_status.go

```diff
@@ -89,6 +105,14 @@ func (s *step2FormatStatus) Execute(ctx *context.Context) error {
 }
 
 func NewGetServiceStatusTask(curveadm *cli.CurveAdm, dc *topology.DeployConfig) (*task.Task, error) {
+	var exectuble string
+	switch dc.GetRole() {
+	case topology.ROLE_ETCD:
+		exectuble = dc.GetRole()
+	default:
+		exectuble = dc.GetKind() + "-" + dc.GetRole()
+	}
+	cmd := "lsof -i -P -n -sTCP:LISTEN " + "/" + dc.GetKind() + "/" + dc.GetRole() + "/sbin/" + exectuble + " | awk '{print$9}' | awk -F ':' '{print$2}'"
```

Comment:
we don't need to run this command when the verbose flag isn't set, otherwise the command will become slow. this means we only get ports when verbose flag is set(curveadm status -v).

Reply:
I added an additional parameter for `NewGetServiceStatusTask` function to detect the value of verbose flag, I also change the docker exec command to `ss`. The performance has improved.

**When verbose mode is enable**:
Before(use `lsof` command):
```
curve@czy-Inspiron-5680:~/curveadm$ time ~/curveadm/sbin/curveadm status -v 
Get Service Status: [OK]  

cluster name    : czy
cluster kind    : curvefs
cluster mds addr: 127.0.0.1:6700(leader),127.0.0.1:6701,127.0.0.1:6702

Id            Role        Host       Replica  Container Id  Status      Listen Port  Log Dir                               Data Dir                            
--            ----        ----       -------  ------------  ------      -----------  -------                               --------                            
9cd171cfd2c1  etcd        127.0.0.1  1/1      41395b7765b6  Up 8 hours  23790,23800  /home/curve/curvefs/logs/etcd0        /home/curve/curvefs/data/etcd0      
6235981e6bed  etcd        127.0.0.1  1/1      e4fe54564136  Up 8 hours  23791,23801  /home/curve/curvefs/logs/etcd1        /home/curve/curvefs/data/etcd1      
b826c5034f47  etcd        127.0.0.1  1/1      37ce2674c1fb  Up 8 hours  23792,23802  /home/curve/curvefs/logs/etcd2        /home/curve/curvefs/data/etcd2      
58ca3c69f1cf  mds         127.0.0.1  1/1      1ac65ce349c0  Up 8 hours  6700,7700    /home/curve/curvefs/logs/mds0         /home/curve/curvefs/data/mds0       
c9be59c37f64  mds         127.0.0.1  1/1      1b99ab279ab4  Up 8 hours  7701         /home/curve/curvefs/logs/mds1         /home/curve/curvefs/data/mds1       
c2b6ab07776a  mds         127.0.0.1  1/1      051a0e51ce70  Up 8 hours  7702         /home/curve/curvefs/logs/mds2         /home/curve/curvefs/data/mds2       
b0b65cfea3e4  metaserver  127.0.0.1  1/1      9a85e96a0647  Up 4 hours  6800,7800    /home/curve/curvefs/logs/metaserver0  /home/curve/curvefs/data/metaserver0
eb9b3c91b4a6  metaserver  127.0.0.1  1/1      9232afa493c4  Up 4 hours  6801,7801    /home/curve/curvefs/logs/metaserver1  /home/curve/curvefs/data/metaserver1
58fd6aa8b68d  metaserver  127.0.0.1  1/1      d2cc8e04ad88  Up 4 hours  6802,7802    /home/curve/curvefs/logs/metaserver2  /home/curve/curvefs/data/metaserver2

real	0m0.854s
user	0m0.036s
sys	0m0.037s
```

After(use `ss` command):
```
curve@czy-Inspiron-5680:~/curveadm$ time ~/curveadm/sbin/curveadm status -v 
Get Service Status: [OK]  

cluster name    : czy
cluster kind    : curvefs
cluster mds addr: 127.0.0.1:6700(leader),127.0.0.1:6701,127.0.0.1:6702

Id            Role        Host       Replica  Container Id  Status      Listen Port  Log Dir                               Data Dir                            
--            ----        ----       -------  ------------  ------      -----------  -------                               --------                            
9cd171cfd2c1  etcd        127.0.0.1  1/1      41395b7765b6  Up 8 hours  23790,23800  /home/curve/curvefs/logs/etcd0        /home/curve/curvefs/data/etcd0      
6235981e6bed  etcd        127.0.0.1  1/1      e4fe54564136  Up 8 hours  23791,23801  /home/curve/curvefs/logs/etcd1        /home/curve/curvefs/data/etcd1      
b826c5034f47  etcd        127.0.0.1  1/1      37ce2674c1fb  Up 8 hours  23792,23802  /home/curve/curvefs/logs/etcd2        /home/curve/curvefs/data/etcd2      
58ca3c69f1cf  mds         127.0.0.1  1/1      1ac65ce349c0  Up 8 hours  6700,7700    /home/curve/curvefs/logs/mds0         /home/curve/curvefs/data/mds0       
c9be59c37f64  mds         127.0.0.1  1/1      1b99ab279ab4  Up 8 hours  7701         /home/curve/curvefs/logs/mds1         /home/curve/curvefs/data/mds1       
c2b6ab07776a  mds         127.0.0.1  1/1      051a0e51ce70  Up 8 hours  7702         /home/curve/curvefs/logs/mds2         /home/curve/curvefs/data/mds2       
b0b65cfea3e4  metaserver  127.0.0.1  1/1      9a85e96a0647  Up 4 hours  6800,7800    /home/curve/curvefs/logs/metaserver0  /home/curve/curvefs/data/metaserver0
eb9b3c91b4a6  metaserver  127.0.0.1  1/1      9232afa493c4  Up 4 hours  6801,7801    /home/curve/curvefs/logs/metaserver1  /home/curve/curvefs/data/metaserver1
58fd6aa8b68d  metaserver  127.0.0.1  1/1      d2cc8e04ad88  Up 4 hours  6802,7802    /home/curve/curvefs/logs/metaserver2  /home/curve/curvefs/data/metaserver2

real	0m0.565s
user	0m0.030s
sys	0m0.049s
```
**When verbose mode is disable**:
my version:
```
curve@czy-Inspiron-5680:~/curveadm$ time ~/curveadm/sbin/curveadm status 
Get Service Status: [OK]  

cluster name    : czy
cluster kind    : curvefs
cluster mds addr: 127.0.0.1:6700(leader),127.0.0.1:6701,127.0.0.1:6702

Id            Role        Host       Replica  Container Id  Status    
--            ----        ----       -------  ------------  ------    
9cd171cfd2c1  etcd        127.0.0.1  1/1      41395b7765b6  Up 8 hours
6235981e6bed  etcd        127.0.0.1  1/1      e4fe54564136  Up 8 hours
b826c5034f47  etcd        127.0.0.1  1/1      37ce2674c1fb  Up 8 hours
58ca3c69f1cf  mds         127.0.0.1  1/1      1ac65ce349c0  Up 8 hours
c9be59c37f64  mds         127.0.0.1  1/1      1b99ab279ab4  Up 8 hours
c2b6ab07776a  mds         127.0.0.1  1/1      051a0e51ce70  Up 8 hours
b0b65cfea3e4  metaserver  127.0.0.1  1/1      9a85e96a0647  Up 4 hours
eb9b3c91b4a6  metaserver  127.0.0.1  1/1      9232afa493c4  Up 4 hours
58fd6aa8b68d  metaserver  127.0.0.1  1/1      d2cc8e04ad88  Up 4 hours

real	0m0.398s
user	0m0.044s
sys	0m0.022s
```
official version:
```
curve@czy-Inspiron-5680:~/curveadm$ time curveadm status 
Get Service Status: [OK]  

cluster name    : czy
cluster kind    : curvefs
cluster mds addr: 127.0.0.1:6700(leader),127.0.0.1:6701,127.0.0.1:6702

Id            Role        Host       Replica  Container Id  Status    
--            ----        ----       -------  ------------  ------    
9cd171cfd2c1  etcd        127.0.0.1  1/1      41395b7765b6  Up 8 hours
6235981e6bed  etcd        127.0.0.1  1/1      e4fe54564136  Up 8 hours
b826c5034f47  etcd        127.0.0.1  1/1      37ce2674c1fb  Up 8 hours
58ca3c69f1cf  mds         127.0.0.1  1/1      1ac65ce349c0  Up 8 hours
c9be59c37f64  mds         127.0.0.1  1/1      1b99ab279ab4  Up 8 hours
c2b6ab07776a  mds         127.0.0.1  1/1      051a0e51ce70  Up 8 hours
b0b65cfea3e4  metaserver  127.0.0.1  1/1      9a85e96a0647  Up 4 hours
eb9b3c91b4a6  metaserver  127.0.0.1  1/1      9232afa493c4  Up 4 hours
58fd6aa8b68d  metaserver  127.0.0.1  1/1      d2cc8e04ad88  Up 4 hours

real	0m0.385s
user	0m0.033s
sys	0m0.042s
```

However, the change breaks the consistency for `NewxxxTask` function(see https://github.com/opencurve/curveadm/pull/123/commits/9f986e54b07dbd0a1cbbd8606e37e1092ef5e861#diff-c3c15742ff739f5546ec2ee2ec864e960753ea1410fe80e07a894a2c4a6eb9edL225-R233)

Please let me know if this works and if you have any suggestions or comments.

Reply:
@Wine93 please check the new change about 'verbose'.

Reply:
> @Wine93 please check the new change about 'verbose'.

I will pick this feature to curveadm `v1.0`.